### ISSUE TITLE
Do not use arrays in appsettings

### ISSUE THREAD

**Issue by jwillmer** [elsa-workflows/elsa-core#1210]
This post covers the issue well: [Avoiding ASP.Net Core Configuration Pitfalls](https://rimdev.io/avoiding-aspnet-core-configuration-pitfalls-with-array-values/)

Basically if we like to use docker to host the application we supply the appsettings as environment variables. However when providing settings for arrays we need to specify the exact array position. That means we will override the existing setting at this location in the default appsettings. Worse, we might override an existing setting that we need and not get our option applied after all since later in the array the original options for our setting was set and will override the setting in our position.

![image](https://user-images.githubusercontent.com/1503577/124434838-a50d7700-dd74-11eb-92e6-1f0a83410939.png)

**Comment by sfmskywalker** [comment]
Ha, good point.

Perhaps we should change from array to dictionary (simple JSON with key/value pairs) where the key is the feature and the value is a Boolean flag (on or off) - which has a potentially added benefit of allowing additional settings to be supplied as part of the feature.

Sample:

```json
{
   "Features": {
        "Persistence.MongoDb": {
            "Enabled": "true",
            "ConnectionStringName": "MongoDb",
            
            // Or specify the connection string inline:
            // "ConnectionString": "mongo://localhost" 
        }
   }
}
```

**Comment by tomy2105** [comment]
Or maybe always expect to have persistence in configuration and have property that determines persistence store.....

{
   "Features": {
        "Persistence": {
            "Type": "MongoDb" // or "SQLServer" or "Postgres" or "SQLite"......
            "ConnectionStringName": "MongoDb",
            
            // Or specify the connection string inline:
            // "ConnectionString": "mongo://localhost" 
        }
   }
}

I like ConnectionStringName here since it is kind of standard to have connection strings in top level configuration......

**Comment by sfmskywalker** [comment]
@tomy2105 

Although that structure looks nice, the tricky part is that then it's no longer generic. For instance, your "Persistence" key becomes a feature. But then you have to associate the `"Persistence"` feature with all persistence provider's Startup classes and have each one of them check the `"Type"` field. It would work, but maybe a little bit too tedious for each provider to be aware of.

Easier to allow each feature implementation to be unaware of other / shared features I think.

**Comment by sfmskywalker** [comment]
Fixed in 20c4df0d148c0e307bda693b49c71d57e89b1b1d
